c7n/resources/rds.py

```diff
@@ -209,6 +177,38 @@ def __call__(self, rdb):
             self.default_vpc = vpcs.pop()
         return vpc_id == self.default_vpc and True or False
 
+@filters.register('kms-alias')
+class KmsKeyAlias(ValueFilter):
+
+    schema = type_schema('kms-alias', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+
+        def _user_kms_alias(resource):
+            client = local_session(self.manager.session_factory).client('kms')
+            key_aliases = client.list_aliases()['Aliases']
```

Comment:
your refetching constants, those should be fetched once and reused, also why the regex instead of contains match? the compile here is superflous since your not reusing the regex. also your inner loop condition allows you to skip the entire check, reorder aka 'KmsKeyId' in resource, if its not in there why are we looping through aliases.